src/cat.ts

```diff
@@ -100,6 +115,38 @@ export class Cat {
     return this._zetas;
   }
 
+  public get prior() {
+    return this._prior;
+  }
+
+  private static validatePrior(priorDist: string, priorPar: number[], minTheta: number, maxTheta: number) {
+    if (priorDist !== 'norm' && priorDist !== 'unif') {
+      throw new Error('The prior distribution you provided is not supported');
+    }
+    if (priorDist === 'norm') {
+      if (priorPar.length !== 2) {
+        throw new Error('The prior distribution parameters you provided are not valid');
+      }
+      if (priorPar[1] <= 0) {
+        throw new Error('The prior distribution standard deviation you provided is not valid');
+      }
+      if (priorPar[0] < minTheta || priorPar[0] > maxTheta) {
+        throw new Error('The prior distribution mean you provided is not valid');
+      }
+    }
+    if (priorDist === 'unif') {
+      if (priorPar.length !== 2) {
+        throw new Error('The prior distribution parameters you provided are not valid');
+      }
```

Comment:
See comment above. This could be removed and checked just once for both distributions.
```suggestion
```